File: app/src/main/java/com/slutsenko/cocktailapp/receiver/BatteryStateReceiver.java
package com.slutsenko.cocktailapp.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.BatteryManager;
import android.widget.Toast;


public class BatteryStateReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        int status = intent.getIntExtra(BatteryManager.EXTRA_STATUS, -1);
        boolean isCharging = status == BatteryManager.BATTERY_STATUS_CHARGING ||
                status == BatteryManager.BATTERY_STATUS_FULL;

// How are we charging?
        int chargePlug = intent.getIntExtra(BatteryManager.EXTRA_PLUGGED, -1);
        boolean usbCharge = chargePlug == BatteryManager.BATTERY_PLUGGED_USB;
        boolean acCharge = chargePlug == BatteryManager.BATTERY_PLUGGED_AC;

        Toast.makeText(context, "Charging", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/slutsenko/cocktailapp/receiver/BatteryStateReceiver.kt
package com.slutsenko.cocktailapp.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent


class BatteryStateReceiver (private val batteryListener: BatteryListener): BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        batteryListener.onBatteryChange(intent!!)
    }

    interface BatteryListener{
        fun onBatteryChange(intent: Intent)
    }
}